src/java.base/unix/classes/sun/nio/fs/UnixFileSystem.java

```diff
@@ -893,7 +893,9 @@ void move(UnixPath source, UnixPath target, CopyOption... options)
             sourceAttrs = UnixFileAttributes.get(source, false);
             if (sourceAttrs.isDirectory()) {
                 // ensure source can be moved
-                access(source, W_OK);
+                int errno = access(source, W_OK);
+                if (errno != 0)
+                    throw new UnixException(errno);
```

Comment:
I don't think it's necessary to throw the UnixException here, should be able to use new UnixException(errno).rethrowAsIOException(source) to have it only throw the appropriate IOException for the error.